spyder/plugins/ipythonconsole/utils/kernelspec.py

```diff
@@ -106,7 +106,7 @@ def argv(self):
             # the activation scripts at spyder/plugins/ipythonconsole/scripts/
             kernel_cmd = [
                 get_activation_script(),  # This is bundled with Spyder
-                get_conda_activation_script(),
+                get_conda_activation_script(pyexec),
```

Comment:
why do we need this change 🙃 ?

Reply:
If `pyexec` is not provided, then the conda activation script associated with Spyder's launch environment is searched for (specifically `sys.prefx`). This is a problem if Spyder is not started from a conda environment (Mac app); but also, more generally, this will ensure that the conda activation script associated with the desired python executable is returned. For example, if Spyder is started from a conda environment, but the provided interpreter is from an environment in a miniconda installation, the miniconda activation script should be returned, not the anaconda one.

Reply:
Understood